packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -272,14 +297,25 @@ class UserSessionManager implements IUserSessionManager {
     value: Nullable<ApiObject> | Nullable<string> | undefined,
   ) {
     const entries = state.storage.entries.value;
-    const storage = entries[sessionKey]?.type as StorageType;
-    const key = entries[sessionKey]?.key as string;
-    if (isStorageTypeValidForStoringData(storage)) {
+    const storageType = entries[sessionKey]?.type as StorageType;
+    if (isStorageTypeValidForStoringData(storageType)) {
       const curStore = this.storeManager?.getStore(
-        storageClientDataStoreNameMap[storage] as string,
+        storageClientDataStoreNameMap[storageType] as string,
       );
+      const key = entries[sessionKey]?.key as string;
       if ((value && isString(value)) || isNonEmptyObject(value)) {
-        curStore?.set(key, value);
+        // if useServerSideCookie load option is set to true
+        // set the cookie from server side
+        if (state.loadOptions.value.useServerSideCookie && storageType === 'cookieStorage') {
+          const encryptedCookieValue = curStore?.encrypt(
+            stringifyWithoutCircular(value, false, [], this.logger),
+          );
+          if (encryptedCookieValue) {
+            this.setServerSideCookie(key, encryptedCookieValue);
+          }
+        } else {
+          curStore?.set(key, value);
+        }
       } else {
         curStore?.remove(key);
```

Comment:
Even this needs to be handled for the server side.

Reply:
This will be handled in the service. If the value is empty string or empty object, the service will set the expireAt/maxage to past time or negative values so that the cookies will be removed.

Reply:
How? The `if` condition on Ln 356 will only allow requests with valid values.

Reply:
Addressed. Now the SDK will send empty string if the value is null, undefined, empty string or empty object.